subworkflows/local/scaffold_curation/main.nf

```diff
@@ -0,0 +1,331 @@
+include { SAMTOOLS_FAIDX                        } from "$projectDir/modules/nf-core/samtools/faidx/main"
+include { SAMTOOLS_MERGE as SAMTOOLS_MERGE_HIC  } from "$projectDir/modules/nf-core/samtools/merge/main"
+include { SAMTOOLS_MERGE as SAMTOOLS_MERGE_HIFI } from "$projectDir/modules/nf-core/samtools/merge/main"
+include { SAMTOOLS_SORT                         } from "$projectDir/modules/nf-core/samtools/sort/main"
+include { BAM2BED_SORT                          } from "$projectDir/modules/local/hic_curation/bam2bed_sort"
+include { COOLER_CLOAD                          } from "$projectDir/modules/nf-core/cooler/cload/main"
+include { COOLER_ZOOMIFY                        } from "$projectDir/modules/nf-core/cooler/zoomify/main"
+include { CREATE_CHROMOSOME_SIZES_FILE          } from "$projectDir/modules/local/hic_curation/create_chromosome_sizes"
+include { PRETEXTMAP                            } from "$projectDir/modules/nf-core/pretextmap/main"   
+include { MINIMAP2_INDEX                        } from "$projectDir/modules/nf-core/minimap2/index/main"
+include { MINIMAP2_ALIGN                        } from "$projectDir/modules/nf-core/minimap2/align/main"
+include { BAM2COVERAGE_TRACKS                   } from "$projectDir/modules/local/hic_curation/bam2coverageTracks"
+include { SEQTK_CUTN                            } from "$projectDir/modules/nf-core/seqtk/cutn/main"
+include { CREATE_GAP_TRACKS                     } from "$projectDir/modules/local/hic_curation/create_gap_tracks"
+include { TIDK_SEARCH as TIDK_SEARCH_BEDGRAPH   } from "$projectDir/modules/nf-core/tidk/search/main"
+include { TIDK_SEARCH as TIDK_SEARCH_TSV        } from "$projectDir/modules/nf-core/tidk/search/main"
+include { TIDK_PLOT                             } from "$projectDir/modules/nf-core/tidk/plot/main"
+include { CREATE_TELOMER_HITILE_TRACK           } from "$projectDir/modules/local/hic_curation/create_telomer_hitile_track"
+include { PRETEXT_TRACKS_INGESTION              } from "$projectDir/modules/local/hic_curation/pretext_tracks_ingestion"
+
+workflow SCAFFOLD_CURATION {
+    take:
+    ch_assemblies // [ meta, assembly ]
+    ch_hic        // [ meta, hic-pairs ]
+    ch_hifi       // [ meta, hifi-reads ]
+
+    main:
+
+    ch_versions  = Channel.empty()
+
+    BWAMEM2_INDEX ( getPrimaryAssembly(ch_assemblies) )
+    ch_versions  = ch_versions.mix( BWAMEM2_INDEX.out.versions )
+
+    SAMTOOLS_FAIDX (
+        getPrimaryAssembly(ch_assemblies),
+        [ [ ], [ ] ]
+    )
+    ch_versions  = ch_versions.mix( SAMTOOLS_FAIDX.out.versions )
+
+    combineByMetaKeys( // Combine (Hi-C + index) with Assembly
+        combineByMetaKeys( // Combine Hi-C reads with BWA index
+            ch_hic, BWAMEM2_INDEX.out.index,
+            keySet: ['id','sample'],
+            meta: 'merge'
+        ),
+        getPrimaryAssembly( ch_assemblies ),
+        keySet: ['id','sample'],
+        meta: 'merge'
+    ).transpose(by:1)
```

Comment:
Add a comment here on the tuple structure, so we know what the `by:1` refers to.